crwatcher/watcher.go

```diff
@@ -152,6 +181,18 @@ func (cw *CRWatcher) setupController() {
 	)
 }
 
+// hasFilter checks to see if we are using an opt in filter (if not, then return true), and if so returns whether we
+// have an annotation matching the given filter.
```

Comment:
This documentation is incorrect. We don't check to see if we are using a filter in here and return true. Should we be? Also based on my other commends can we make this `clearsFilter` and it also checks if there is even a filter in play.

Reply:
Changed to clearsFilter, and the docs are changed accordingly.